src/mono/msbuild/apple/data/ProxyProjectForAOTOnHelix.proj

```diff
@@ -99,10 +107,22 @@
       <_RuntimePackFiles Condition="%(_AppleUsedRuntimePackFiles.Extension) != '.dll' or %(_AppleUsedRuntimePackFiles.FileName) == 'System.Private.CoreLib'" Include="@(_AppleUsedRuntimePackFiles->'$(MicrosoftNetCoreAppRuntimePackNativeDir)%(FileName)%(Extension)')" />
     </ItemGroup>
 
+    <ItemGroup>
+      <ReferencePath Include="@(AppleReferenceSharedPathFiles->'$(MicrosoftNetCoreAppRuntimePackLibDir)%(FileName)%(Extension)')" />
+      <ReferencePath Include="@(AppleReferenceExtraPathFiles->'$(ExtraFilesPath)%(FileName)%(Extension)')" />
+    </ItemGroup>
+
     <Copy SourceFiles="@(_RuntimePackFiles)"
           DestinationFolder="$(OriginalPublishDir)" />
   </Target>
 
+  <Target Name="_TrimApp" DependsOnTargets="ILLink;CopyFilesToPublishDirectory">
+    <ItemGroup>
+      <AppleAssembliesToBundle Remove="@(AppleAssembliesToBundle)" />
```

Comment:
As we are only adjusting the `AppleAssembliesToBundle` item group, I don't think it is necessary to have a dedicated build target for that. 
Although, it is possible there isn't a better place to do so. 
In any case, the name does not seem suitable (something like `_AdjustTrimmedAssembliesToBundle` might be a better option, but I don't have a strong opinion).

Reply:
Some additional thoughts..

We reached a point where we managed to enable trimming on Helix and things work, which is great as it was much needed and long awaited feature. However, some parts of this implementation could be improved in general. 

One of the core problems was splitting up the build/publish across build machines and Helix.  That involved:
- recreating the build folder structure on Helix, but also 
- configuring the build project "to resume" app trimming and deployment.

The problem that remains - which we did manage to workaround - is that we are reusing the `publish` folder for different things and rewriting it during the build on Helix. That is not ideal, will be hard to maintain and should be improved in the future. 
Shuffling `AppleAssembliesToBundle` here is done for this reason, we have an initial set of assemblies to bundle, but during trimming some might be removed, so we have to manually adjust this item group. 

Another issue with reusing the publish folder for everything is that we place the `ProxyProjectForAOTOnHelix.props`, `ProxyProjectForAOTOnHelix.proj` and many other files in it, so we can end up bundling them into the final app bundle, which is again unnecessary and error prone. It is true we have some file filtering in `AppleAppBuilder` task to prevent bundling unnecessary files, but that shouldn't be the place to fix these kind of problems. 

I propose we handle these improvements as a follow-up task (and not delay this PR further) but we should make sure we have a tracking issue for it. @kotlarmilos what do you think?

PS For now, I can also add a comment in the code to explain the reason for `_AdjustTrimmedAssembliesToBundle`

Reply:
Add a task in the linked issue and a comment describing what is the purpose of `_AdjustTrimmedAssembliesToBundle`